skbio/sequence/_sequence.py

```diff
@@ -538,9 +586,13 @@ def _assert_can_cast_to(cls, target):
     def _positional_metadata_axis_len_(self):
         return len(self)
 
+    @overrides(IntervalMetadataMixin)
+    def _interval_metadata_axis_len_(self):
+        return len(self)
+
     @stable(as_of="0.4.0")
     def __init__(self, sequence, metadata=None, positional_metadata=None,
-                 lowercase=False):
+                 interval_metadata=None, lowercase=False):
```

Comment:
This is stable API so I don't think the parameter order should be changed. @ebolyen, IIRC you argued strongly earlier this year to avoid stable API changes (unless it results in fixing a bug) so I think we should hold to that here.

It obviously is more natural to have it immediately after positional metadata but will break code that isn't using keyword arguments. Too bad they aren't defined as py3 keyword-only args.